test/api/test_catalog.py

```diff
@@ -250,22 +262,44 @@ def assert_metadata_record(schema_id):
         assert metadata_record['schema_uri'] == schema_uris[schema_id]
 
         # construct the expected metadata
-        expected_metadata = metadata_examples[schema_id]
+        expected_metadata = deepcopy(metadata_examples[schema_id])
         if schema_id == 'SchemaOrg.Dataset':
             expected_metadata['identifier'] = f'doi:{example_record.doi}' if example_record.doi else None
-            if has_iso19115:
-                expected_metadata['keywords'] = [
-                    dk['keyword'] for dk in metadata_examples['SAEON.ISO19115']['descriptiveKeywords']
-                    if dk['keywordType'] in ('general', 'place', 'stratum')
-                ]
-            else:
-                expected_metadata['keywords'] = [
-                    s['subject'] for s in metadata_examples['SAEON.DataCite4']['subjects']
-                ]
+            expected_metadata['keywords'] = iso_keywords if has_iso19115 else generic_keywords
             expected_metadata['@id'] = expected_metadata['url'] = (
                 'http://odp.catalog/mims/'
                 f'{example_record.doi if example_record.doi else example_record.id}'
             )
+        elif schema_id == 'RIS.Citation':
+            expected_metadata['ris'] = expected_metadata['ris'].split('\n')
+            metadata_record['metadata']['ris'] = metadata_record['metadata']['ris'].split('\n')
+
+            metadata_record_doi = None
+            metadata_record_url = None
+            for item in metadata_record['metadata']['ris']:
+                if "DO -" in item:
+                    metadata_record_doi = item
+                elif "UR -" in item:
+                    metadata_record_url = item
+
+            for item in expected_metadata['ris']:
+                if "DO -" in item:
+                    expected_metadata['ris'].remove(item)
+                    if metadata_record_doi:
+                        expected_metadata['ris'].append(metadata_record_doi)
+
+                if "UR -" in item:
+                    expected_metadata['ris'].remove(item)
+                    if metadata_record_url:
+                        expected_metadata['ris'].append(metadata_record_url)
```

Comment:
This is an improvement in that the test is now modifying the expected metadata rather than the actual API result. There are however two subtle issues that I think it would be worth addressing:

* The expected metadata should be constructed based on the information we already have prior to getting the result from the API, i.e. from info in the example record. Otherwise it's possible that we might miss some kind of systematic error where the API is returning an incorrect DOI or URL. So instead of variables `metadata_record_doi|url`, rather set, say, `expected_doi|url` variables based on `example_record.doi` and similarly to how `expected_metadata['url']` is set in the SchemaOrg.Dataset block just above.
* The second loop is modifying the list that is being iterated over. If you watch the value of `expected_metadata['ris']` while stepping through the iterations you'll see that the `DO -` and `UR -` items may be removed and added several times. You can get around this by iterating over a shallow copy of the list (`from copy import copy`).